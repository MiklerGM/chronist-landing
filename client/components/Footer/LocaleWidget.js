import React from 'react';
import PropTypes from 'prop-types';
import ym from 'react-yandex-metrika';

class LocaleWidget extends React.Component {
  changeToEn() {
    this.props.onChangeLanguage('en');
  }

  changeToRu() {
    this.props.onChangeLanguage('ru');
  }

  toggleLocale() {
    if (this.props.locale === 'en') {
      this.props.onChangeLanguage('ru');
    } else {
      this.props.onChangeLanguage('en');
    }
  }

  render() {
    if (!this.props.min) {
      return (
        <p>
          <button
            onClick={() => this.changeToEn()}
            className='decorless locale--toggle'
            style={(this.props.locale === 'en') ? { color: '#ffffff' } : null}
          >
            English
          </button>
          { '|' }
          <button
            onClick={() => this.changeToRu()}
            className='decorless locale--toggle'
            style={(this.props.locale === 'ru') ? { color: '#ffffff' } : null}
          >
            Русский
          </button>
        </p>
      );
    }
    return (
      <a
        onClick={() => this.toggleLocale()}
        // className='decorless'
      >
        {this.props.locale === 'en' ? 'RU' : 'EN'}
      </a>
    );
  }
}

LocaleWidget.propTypes = {
  locale: PropTypes.string.isRequired,
  onChangeLanguage: PropTypes.func.isRequired
};

export default LocaleWidget;
