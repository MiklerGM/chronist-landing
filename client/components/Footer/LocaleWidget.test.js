import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import LocaleWidget from './LocaleWidget';

// smoke test. it rendering without warnings

it('locale widget with en without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<LocaleWidget onChangeLanguage={e => console.log('dump func', e)} locale='en' />, div);
});

it('locale widget with ru without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<LocaleWidget onChangeLanguage={e => console.log('dump func', e)} locale='ru' />, div);
});


// Snapshot tests
it('renders correctly', () => {
  const tree = renderer
    .create(<LocaleWidget onChangeLanguage={e => console.log('dump func', e)} locale='ru' />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
