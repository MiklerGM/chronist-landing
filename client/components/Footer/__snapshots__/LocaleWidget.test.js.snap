// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p>
  <button
    className="decorless locale--toggle"
    onClick={[Function]}
    style={null}
  >
    English
  </button>
  |
  <button
    className="decorless locale--toggle"
    onClick={[Function]}
    style={
      Object {
        "color": "#ffffff",
      }
    }
  >
    Русский
  </button>
</p>
`;
